packages/lit-dev-content/.eleventy.js

```diff
@@ -412,6 +424,12 @@ ${content}
       // them directly instead of spending its own cycles. Note this adds ~4
       // seconds to the build, but it's disabled during dev.
       await preCompress({glob: `${OUTPUT_DIR}/**/*`});
+
+      await fs.writeFile(
+        path.join(OUTPUT_DIR, 'csp-inline-script-hashes.txt'),
```

Comment:
Could this `writeFile` somehow be contributing to the timeout in the integration test? The github action consistently fails in the changes off this one.

Reply:
Hmmm I got the error randomly while manually testing. It had a could more details that the github actions truncated.

```shell
Template render error was thrown:
...
...
EADDRINUSE: address already in use :::8001.
```

Reply:
> File is written to `OUTPUT_DIR`, and later the server reads from `path.join(contentPackage, '_site', 'csp-inline-script-hashes.txt'),`. Is this something only scoped to production builds?

Yep, this doesn't happen in dev mode. Added comment explaining that we don't use hashes in dev mode, because we don't inline scripts.

> Could `writeFile` somehow be contributing to the timeout in the integration test? The github action consistently fails in the changes off this one.

I *think* it's unlikely to be related to the new `writeFile`. The new `writeFile` blocks the end of the Eleventy build process, so testing couldn't start until this write is done. And I can't think why a pending file write would cause this effect anyway. The server would fail to start if the hash file didn't exist yet.

> Hmmm I got the error randomly while manually testing. It had a couple more details that github actions truncated.
> 
> ```shell
> Template render error was thrown:
> ...
> ...
> EADDRINUSE: address already in use :::8001.
> ```

That's a weird error. From the message I would guess it's happening during Eleventy HTML rendering. We do spin up a server as part of the Eleventy build in order to render static code snippets:

https://github.com/lit/lit.dev/blob/44686533f89dc8bbcc28126664d2e7da65470232/packages/lit-dev-tools/src/playground-plugin/renderer.ts#L167

It should be just finding a free port and using it, and we share a single instance of the server for the entire build, so I'm confused about why there would be something else contending to listen on port `8001`... very odd.


> I wonder why there was a write error in the integration test.

Currently I seem to be getting failures on the "Download link for updated screenshots" job -- twice in a row, even though the main "Integration tests" job is succeeding. They seem to be identical jobs, except that one writes out the screenshot artifacts. The errors seem to be timing out waiting for selectors on the page.

```
> lit-dev-server@0.0.0 start:playground
> MODE=playground PORT=6416 node lib/server.js

mode: playground
port: 6416
static root: /home/runner/work/lit.dev/lit.dev/packages/lit-dev-content/node_modules/playground-elements
server listening on port 6416
····T··

  1) playground.spec.ts:10:3 › Playground default example is simple-greeting.ts ====================

    Timeout of 30000ms exceeded.

    Pending operations:
      - page.waitForSelector at tests/playground.spec.ts:74:14

    page.waitForSelector: Protocol error (Runtime.callFunctionOn): Target closed.
    =========================== logs ===========================
    waiting for selector "playground-preview [part="preview-loading-indicator"][aria-hidden="true"]" to be visible
    ============================================================

      72 |
      73 | async function waitForPlaygroundPreviewToLoad(page: Page) {
    > 74 |   await page.waitForSelector(
         |              ^
      75 |     'playground-preview [part="preview-loading-indicator"][aria-hidden="true"]'
      76 |   );
      77 |   // There is a fade-out transition on the playground loading bar that makes

        at waitForPlaygroundPreviewToLoad (/home/runner/work/lit.dev/lit.dev/tests/playground.spec.ts:74:14)
        at /home/runner/work/lit.dev/lit.dev/tests/playground.spec.ts:13:11
        at WorkerRunner._runTestWithBeforeHooks (/home/runner/work/lit.dev/lit.dev/node_modules/@playwright/test/lib/test/workerRunner.js:426:7)

  Slow test: playground.spec.ts (54s)
  Slow test: components-styles.spec.ts (15s)

  1 failed
    playground.spec.ts:10:3 › Playground default example is simple-greeting.ts =====================
  6 passed (1m)
ERROR: "test:integration:update-golden-screenshots" exited with 1.
Error: Process completed with exit code 1.
```

Reply:
> Currently I seem to be getting failures on the "Download link for updated screenshots" job -- twice in a row, even though the main "Integration tests" job is succeeding. They seem to be identical jobs, except that one writes out the screenshot artifacts. The errors seem to be timing out waiting for selectors on the page.

After another retry, both screenshot jobs are passing. Since I can see that things are fine manually, I'm going to keep moving through these PRs and keep an eye on the test failures. I suspect there is some flakiness that's actually just unrelated to this change, and we got unlucky, but you never know -- sometimes very surprising interactions happen.